plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -140,16 +140,29 @@ func (g *genericScheduler) selectHost(priorityList schedulerapi.HostPriorityList
 		return "", fmt.Errorf("empty priorityList")
 	}
 
-	sort.Sort(sort.Reverse(priorityList))
-	maxScore := priorityList[0].Score
-	firstAfterMaxScore := sort.Search(len(priorityList), func(i int) bool { return priorityList[i].Score < maxScore })
+	// extract nodes with highest score
+	// TODO(harry) should we refactor PrioritizeNodes into return nodes with max score only?
+	maxHostList := make(schedulerapi.HostPriorityList, 0)
+	maxPriorityScore := -1
+	for _, node := range priorityList {
+		if node.Score > maxPriorityScore {
+			maxPriorityScore = node.Score
+			// reset the maxHostList
+			maxHostList = schedulerapi.HostPriorityList{node}
```

Comment:
For performance, we shouldn't reallocate it every time we face a better result. What should be better is to start overriding elements from the first one. And then potentially at then end, shrink the result.